### PR TITLE
build: create architecture as a concept in our build scripts

### PR DESCRIPTION
**Pull request by jazzdan**
This paves the way for also building for arm64 on macOS.

Related to RUN-1943

### CODE REVIEW

File: replay_build_scripts/common.mjs [hunk 1]
```diff
@@ -43,6 +43,20 @@ export function currentPlatform() {
   }
 }
 
+export function outputArchitecture() {
```
Comment (jazzdan) [inline replay_build_scripts/common.mjs:46]: This is not the ideal way for us to handle architecture but it's how we effectively do it today.

File: replay_build_scripts/upload_build_artifacts.mjs [hunk 2]
```diff
@@ -18,7 +19,8 @@ const BUILDKITE_BUILD_ID_ARTIFACT = "build_id";
 const BUILDKITE_ARTIFACT_DIRECTORY = path.join(
   process.env.BUILDKITE_BUILD_CHECKOUT_PATH || "./",
   "build_id",
-  currentPlatform()
+  currentPlatform(),
+  outputArchitecture()
```
Comment (klochek) [inline replay_build_scripts/upload_build_artifacts.mjs:23]: You need to update the `scripts/chromium/tests/buildkite_run_tests_from_metadata.sh` in backend for this change, as well as the test pipeline file as well.

Comment (klochek) [reply replay_build_scripts/upload_build_artifacts.mjs:23]: artifact, not metadata, sorry!

Comment (jazzdan) [reply replay_build_scripts/upload_build_artifacts.mjs:23]: Done over here https://github.com/replayio/backend/pull/7738#pullrequestreview-1427395314

**Review by klochek — APPROVED** [review]

**Pull request merged**